pkg/crdutil/crdutil.go

```diff
@@ -49,26 +51,32 @@ func (s *StringList) Set(value string) error {
 }
 
 var (
-	crdsDir StringList
+	files     []string
+	recursive bool
 )
 
 func initFlags() {
-	flag.Var(&crdsDir, "crds-dir", "Path to the directory containing the CRD manifests")
```

Comment:
Are we concerned about this as a breaking change since the command line arguments are changed?

Reply:
I was also thinking about this but this pkg is pretty new and we are only using it for the network-operator currently

Reply:
If the network-operator is not released with this change, better to just update as it will be first use of this package.

Reply:
The network-operator 24.10.0 has been released 2 days ago with the change in it, see https://github.com/Mellanox/network-operator/blob/v24.10.0/deployment/network-operator/templates/upgrade-crd-hook.yaml#L84.

But IMO it is totally fine to change it, because it is a Helm pre-install hook. If it fails, it fails the installation/upgrade and the user has to use the latest Helm chart. There is no dependency to the runtime which can cause the operator to fail.

WDYT @e0ne @rollandf

Reply:
If there is a concern, we could support both -- marking `crds-dir` as deprecated -- for a release. Happy with the change since this is effectively an internal API.

Reply:
We can also revert it to `--crds-dir`. Both `crds-dir` and `filename` does not contain both files and dirs, so it is not 100% explicit anyway :)
Request came through https://github.com/NVIDIA/k8s-operator-libs/pull/58#discussion_r1859239149

Reply:
@shivamerla WDYT? Should we remove the flag rename and recursive flag (see https://github.com/NVIDIA/k8s-operator-libs/pull/58#discussion_r1867228854) out of this PR and change it in a follow-up? I'd say this will reduce the size of this PR.